### PR TITLE
[PLAT-715] CreateRealm keycloak methods

### PR DESCRIPTION
**Pull request by carrala**
- adds methods for creating a realm
- temporarily changes the Register URL in client service so that it can be used with identity-service

this is not ready to be merged in yet, but i'm putting it up for reference
refer to: [https://docs.google.com/document/d/1p2OIXzCwPkhYnGjJC8pq-rv_xLZsFGPfzXtRvutKWBE/edit?usp=sharing](url)

### CODE REVIEW

File: clientServiceClient/clientServiceClient.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (c *ClientServiceClient) AdminToggleClientEnabled(ctx context.Context, para
 // Register registers a client.
 func (c *ClientServiceClient) Register(ctx context.Context, params ClientRegisterRequest) (*ClientRegisterResponse, error) {
 	var result *ClientRegisterResponse
-	path := c.Host + "/" + "v1/account/e3db/clients/register"
+	path := c.Host + "/" + ClientServiceBasePath
```
Comment (rosasck) [inline clientServiceClient/clientServiceClient.go:79]: this should be its own PR, reminder 
`InternalRegister` -- client service 
`Register`-- account

File: keycloakClient/keycloakClient.go [hunk 2]
```diff
@@ -117,6 +122,73 @@ func (c *Client) doTokenRequest(realm string, body string) (*TokenInfo, error) {
...
 
+func setAuthorizationAndHostHeaders(req *http.Request, accessToken string) (*http.Request, error) {
+	host, err := extractHostFromToken(accessToken)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	req.Header.Add("X-Forwarded-Proto", "https")
+	// is this always the content type? fails with 415 if content type isn't specified
+	req.Header.Add("Content-Type", "application/json")
```
Comment (galxy25) [inline keycloakClient/keycloakClient.go:133]: both of these could vary (e.g. for uploading a file it will be a different content type) so need to be parameters that get passed into this function

File: keycloakClient/keycloakClient.go [hunk 3]
```diff
@@ -117,6 +122,73 @@ func (c *Client) doTokenRequest(realm string, body string) (*TokenInfo, error) {
...
+	req, err = setAuthorizationAndHostHeaders(req, accessToken)
+	if err != nil {
+		return "", err
+	}
+	response, err := e3dbClients.ReturnRawServiceCall(c.httpClient, req, nil)
+	if err != nil {
+		return "", e3dbClients.NewError(err.Error(), path, response.StatusCode)
+	}
+	location := response.Header.Get("Location")
+	// Might need to change this to account for different content types
```
Comment (galxy25) [inline keycloakClient/keycloakClient.go:184]: def

File: keycloakClient/keycloakClient_test.go [hunk 4]
```diff
@@ -34,3 +45,61 @@ func TestGetTokenSucceedsWithValidClientCredentials(t *testing.T) {
 		t.Fatal("Token must be nonempty.")
 	}
 }
+
+func TestCreateRealmSucceeds(t *testing.T) {
+	// this is just a random registration token, but can change this to generate a realm one for a specific account
+	registrationToken := "5a764bb8259c90365785da9be32dacf5fcda715d96b07434c92ef46a606ff4be"
```
Comment (galxy25) [inline keycloakClient/keycloakClient_test.go:51]: should come from an environment variable

**Review by galxy25 — APPROVED** [review]
approved pending resolution of comments
